### ISSUE TITLE
axvline is sometimes not shown

### ISSUE THREAD

**Issue by idavydov** [matplotlib/matplotlib#6960]
Usually if you plot histogram + axvline, the x axis limits are extended so the vertical line is visible.

E.g.
![figure_1_good](https://cloud.githubusercontent.com/assets/671660/17806695/52713ba2-6606-11e6-965e-ef8e2d680491.png)

But sometimes the x axis is not extended to be able to show a vertical line.
Here's the code to demonstrate the issue:
```python
from matplotlib import pylab as plt
plt.hist([9138, 9138, 9133, 9137, 9134, 9128, 9133, 9134, 9134, 9142])
plt.axvline(x=9144, color='red')
plt.show()
```

![figure_1_bad](https://cloud.githubusercontent.com/assets/671660/17806715/6d8d3382-6606-11e6-97a9-d1728a54c422.png)

(the red vertical line should be visible on the right, x=9144)

- Matplolib: 1.5.1-1ubuntu1 (from repository), Python 2.7.12, Ubuntu 16.04.1
- The same problem can be reproduced in python 3.5.2.

**Comment by jenshnielsen** [comment]
This is probably fixed in 2.0 betas. What I think is going on is. The current autolim tries to find the best round numbers that exactly contains all data i.e. max(datacoordinates) <= maxlim in this case it is an exact equality and the lim is picked as 9144 which contains the data but renders it invisible. The new default in 2.0 will be maxlim = max(datacoordinates) + somemargin which will solve this problem.

**Comment by LindyBalboa** [comment]
I just checked, this behaves properly on master.

**Comment by idavydov** [comment]
I can confirm that it works properly on the latest master (2cdb5aab4f2a623185182f4e812e120c208bcedc).

**Comment by jenshnielsen** [comment]
Thanks closing then, This should make it into 2.0 and is currently available in the 2.0 beta3. Installable with pip using `pip install matplotlib --pre --upgrade`
